Creating language-specific workspace pages.

gvwilson opened the pull request:
The original comment is out of date - see "I've pushed more changes" below.

gvwilson committed:
Switching languages based on URL or query parameters.

1.  Add `getLanguage` to `index.js` to get a two-letter language code from the URL path or the query parameters.
    (The latter allows us to specify a language when loading `index.html` directly for testing.)
2.  Modify `index.html` to call this function and pass the result to `setup`.

This should allow us to:

1.  Move the current `/index.html` to `/en/index.html`, so that `https://tidyblocks.tech/en/` will do the right thing.
2.  Create `/es/index.html` to provide a Spanish interface.
3.  Put in a new `./index.html` that redirects to `/en/index.html` (or alternatively provide a landing page that allows people to select their preferred language).

DarkMetroid commented:
Looks good, much prefer this approach.

Just to confirm, `/en/index.html` will need to be handled by Jekyll (using a front matter/permalink to do the routing)? To develop locally without Jekyll we'd need to also have an `index.html` without the Jekyll front matter syntax.

I feel like that makes sense, and the duplication in index files there is unavoidable if we want to work locally and do routing via. Jekyll.

I like the idea of a landing page for languages (or maybe we could just integrate it as a dropdown on the menu bar that redirects), just obviously forces us to run Jekyll during development. That doesn't seem like an unreasonable thing to require though.

gvwilson committed:
Merge branch 'master' into languages

gvwilson committed:
Reorganizing website

gvwilson commented:
I've pushed some more changes that reorganize things so we don't need Jekyll on the TidyBlocks interface page:

1.  Each language now has a sub-directory ('en', 'es', etc.).
2. That sub-directory contains a copy of `index.html` (the TidyBlocks interface page localized to that language thanks to the URL containing the language code).
3. I have removed the language guessing, since each language has its own `index.html` and we can just hardwire the language into it.
4. The sub-directory also contains the help files for that language. (Right now these are only present for 'en', but I hope we will get translations.) Note that the help files include the glossary and the FAQ.
5. The top level has the blog (which I doubt we'll ever translate), the license (ditto), the Code of Conduct (maybe?), the list of authors (unlikely). If we _do_ translate any of these files, it will be easy to move them down into language directories.
6. I've adjusted the URLs in the help menu in `libs/ui/menuBar.jsx` to point at the right things using relative paths.

- Note that there is a file `en/guide.md` which seems a bit redundant now, but not having a root page for the documentation felt weird too.
- And the site home page now has links to the English and Spanish versions rather than being the interface itself. This means there's an extra click to get to TidyBlocks, but it puts all languages on an equal footing.

Feedback would be very welcome. cc @DarkMetroid @MayaGans @demar01

gvwilson committed:
Removing language guessing

DarkMetroid commented:
I like the plan, makes sense.
